File: my_two_program_tutorial/python/start-client.py
from splashkit import *

def start_client(name, host='127.0.0.1', port=65432):
    try:
        # Create the client connection
        client = open_connection(name, host, port)
        write_line(f"Connected to server at {host}:{port}")
        
        # Keep the connection open
        while is_connection_open(client):
            check_network_activity()
        
    finally:
        if client:
            # Close the client connection
            close_connection(client)
        write_line("Client closed.")

if __name__ == "__main__":
    start_client("MyClient", "127.0.0.1", 65432)

File: my_two_program_tutorial/c-plus/start-client.cpp
#include "splashkit.h"
#include <string>

void start_client(const std::string& name, const std::string& host = "127.0.0.1", int port = 65432) {
    // Create the client connection
    connection client = open_connection(name, host, port);
    write_line("Connected to server at " + host + ":" + std::to_string(port));

    // Keep the connection open
    while (is_connection_open(client)) {
        check_network_activity();
    }

    // Close the client connection
    close_connection(client);
    write_line("Client closed.");
}

int main() {
    start_client("MyClient", "127.0.0.1", 65432);
    return 0;
}
